src/main/java/seedu/savenus/model/purchase/TimeOfPurchase.java

```diff
@@ -22,6 +22,12 @@ public static TimeOfPurchase generate() {
         return new TimeOfPurchase(TimeStamp.generateCurrentTimeStamp());
     }
 
+    @Override
+    public int compareTo(TimeOfPurchase other) {
+        return getTimeStampInLocalDateTime()
+                .compareTo(other.getTimeStampInLocalDateTime());
+    }
```

Comment:
You can consider implementing the comparator such that other.getTimeStampInLocalDateTime().compareTO(this.getTimeStampInLocalDateTime() instead, this will automatically add your new objects into your PurchaseList in a way that they are arranged in a descending order.

Reply:
Yea that makes sense too. I reversed the list just before displaying, should be fine the way it is I think!